src/modm/driver/encoder/encoder_impl.hpp

```diff
@@ -0,0 +1,46 @@
+#ifndef MODM_ENCODER_HPP
+#error "Don't include this file directly, use 'encoder.hpp' instead!"
+#endif
+
+template<typename PH_A, typename PH_B, modm::encoder::Division_t DIVISION, typename UnderlyingType>
+inline void
+modm::Encoder<PH_A, PH_B, DIVISION, UnderlyingType>::update()
+{
+	// Tare on power-on
+	UnderlyingType raw = 0;
+	UnderlyingType diff;
+
+	auto read = Phases::read();
+	if (read & 0b10) raw = 3;
+	if (read & 0b01) raw ^= 1;  // convert gray to binary
+
+	diff = raw_last - raw;
```

Comment:
This can be declared and initialized on one line instead of declaring it uninitialized:
```suggestion
	const auto diff = raw_last - raw;
```